### PR TITLE
TranslateToCpp: Emit floating point literals with suffix

### PR DESCRIPTION
**Pull request by mgehre-amd**
Emits `2.0e+00f` instead of `(float)2.0e+00`.

This helps consumers of the emitted code, especially when there are large numbers of floating point literals, to have a simple AST.

### CODE REVIEW

File: mlir/lib/Target/Cpp/TranslateToCpp.cpp [hunk 1]
```diff
@@ -1193,10 +1192,18 @@ LogicalResult CppEmitter::emitAttribute(Location loc, Attribute attr) {
 
   // Print floating point attributes.
   if (auto fAttr = dyn_cast<FloatAttr>(attr)) {
+    if (!fAttr.getType().isF32() && !fAttr.getType().isF64()) {
```
Comment (simon-camp) [inline mlir/lib/Target/Cpp/TranslateToCpp.cpp:1195]: Maybe this is more readable (nit).
```suggestion
    if (!isa<Float32Type, Float64Type>(fAttr.getType()) {
```

File: mlir/lib/Target/Cpp/TranslateToCpp.cpp [hunk 2]
```diff
@@ -1193,10 +1192,18 @@ LogicalResult CppEmitter::emitAttribute(Location loc, Attribute attr) {
...
+      return emitError(loc,
+                       "expected floating point attribute to be f32 or f64");
+    }
     printFloat(fAttr.getValue());
     return success();
   }
   if (auto dense = dyn_cast<DenseFPElementsAttr>(attr)) {
+    if (!dense.getElementType().isF32() && !dense.getElementType().isF64()) {
```
Comment (simon-camp) [inline mlir/lib/Target/Cpp/TranslateToCpp.cpp:1203]: Same as above

File: mlir/test/Target/Cpp/const.mlir [hunk 3]
```diff
@@ -23,10 +23,10 @@ func.func @emitc_constant() {
 // CPP-DEFAULT-NEXT: uint8_t [[V4:[^ ]*]] = 255;
 // CPP-DEFAULT-NEXT: char [[V5:[^ ]*]] = CHAR_MIN;
 // CPP-DEFAULT-NEXT: size_t [[V6:[^ ]*]] = 2;
-// CPP-DEFAULT-NEXT: float [[V7:[^ ]*]] = (float)2.000000000e+00;
+// CPP-DEFAULT-NEXT: float [[V7:[^ ]*]] = 2.000000000e+00f;
```
Comment (simon-camp) [inline mlir/test/Target/Cpp/const.mlir:26]: Can you insert an f64 example. Seems we never had a test for doubles.

**Review by simon-camp — CHANGES_REQUESTED** [review]
Looks good. I have just two small suggestions.

File: mlir/test/Target/Cpp/const.mlir (hunk 3 above)
Comment (mgehre-amd) [reply mlir/test/Target/Cpp/const.mlir:26]: Sure, will do

**Review by simon-camp — APPROVED** [review]

**Review by marbre — APPROVED** [review]

**Pull request merged**